src/NuGet.Core/NuGet.Packaging/PackageReaderBase.cs

```diff
@@ -542,5 +545,54 @@ public virtual Task<string> CopyNupkgAsync(string nupkgFilePath, CancellationTok
         {
             throw new NotImplementedException();
         }
+
+        public async Task<IReadOnlyList<Signature>> GetSignaturesAsync(CancellationToken token)
+        {
+            var signatures = new List<Signature>();
+
+            var files = await GetFilesAsync(token);
```

Comment:
I'd like for signing and signature verification API's to not rely on [implicit percent-decoding of ZIP entry paths](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/PackageExtraction/ZipArchiveExtensions.cs#L33).

Ideally, this line and the next could boil down to:

    var sigFile = zipArchive.GetEntry("testsigned/signed.json");

...and similarly simplify the `IsSignedAsync(...)` method.

Reply:
I'll make these abstract and move this to the higher level implementations where zip archive can be used directly.